static/app/views/onboarding/useBackActions.tsx

```diff
@@ -0,0 +1,206 @@
+import {useCallback, useContext} from 'react';
+import {useBlocker} from 'react-router-dom';
+import type {useAnimation} from 'framer-motion';
+
+import {removeProject} from 'sentry/actionCreators/projects';
+import {OnboardingContext} from 'sentry/components/onboarding/onboardingContext';
+import type {Project} from 'sentry/types/project';
+import {defined} from 'sentry/utils';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import {handleXhrErrorResponse} from 'sentry/utils/handleXhrErrorResponse';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import {onboardingSteps} from 'sentry/views/onboarding/onboarding';
+import {hasDocsOnPlatformClickEnabled} from 'sentry/views/onboarding/platformSelection';
+import type {StepDescriptor} from 'sentry/views/onboarding/types';
+
+/**
+ * Custom hook that performs a couple of actions when navigating back in the onboarding flow,
+ * either by clicking the native browser back button or a custom back button.
+ */
+export function useBackActions({
```

Comment:
Nit, would name it a bit more descriptive as others might get autocompletion for it otherwise.
```suggestion
export function useOnboardingBackActions({
```

Reply:
I wasn't concerned about that since it's in the onboarding folder. They should be able to see it, right?